src/Umbraco.Core/EmbeddedResources/BlockGrid/items.cshtml

```diff
@@ -17,9 +20,17 @@
             style=" --umb-block-grid--item-column-span: @item.ColumnSpan; --umb-block-grid--item-row-span: @item.RowSpan; ">
             @{
                 var partialViewName = "blockgrid/Components/" + item.Content.ContentType.Alias;
+
                 try
                 {
-                    @await Html.PartialAsync(partialViewName, item)
+                    if (_selector.SelectComponent(item.Content.ContentType.Alias) is not null)
+                    {
+                        @await Component.InvokeAsync(item.Content.ContentType.Alias, item)
+                    }
+                    else
+                    {
+                        @await Html.PartialAsync(partialViewName, item)
+                    }
```

Comment:
OK, I see what you mean about the exception, now that I see it I think we need to change this.

We don't want to catch the exception thrown from the component logic, it's already got a check and any `InvalidOperationException` will be for reasons that should be thrown/surfaced properly. Also, the message would be wrong.

Something like:

```csharp
if (_selector.SelectComponent(item.Content.ContentType.Alias) is not null)
{
    @await Component.InvokeAsync(item.Content.ContentType.Alias, item)
}
else
{
    var partialViewName = "blockgrid/Components/" + item.Content.ContentType.Alias;

    try
    {
        @await Html.PartialAsync(partialViewName, item)
    }
    catch (InvalidOperationException)
    {
        <p>
            <strong>Could not render component of type: @(item.Content.ContentType.Alias)</strong>
            <br />
            This likely happened because the partial view <em>@partialViewName</em> could not be found.
        </p>
    }
}
```

Reply:
You are right, I have updated that as well :)